jerry-debugger/src/lib/debugger-client.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import WebSocket from 'ws';
+
+export interface JerryDebuggerOptions {
+  host?: string;
+  port?: number;
+  verbose?: boolean;
+}
+
+export const DEFAULT_DEBUGGER_HOST = 'localhost';
+export const DEFAULT_DEBUGGER_PORT = 5001;
+
+export class JerryDebugger {
+  readonly host: string;
+  readonly port: number;
+  readonly verbose: boolean;
+  private socket?: WebSocket;
+
+  constructor(options: JerryDebuggerOptions) {
+    this.host = options.host || DEFAULT_DEBUGGER_HOST;
+    this.port = options.port || DEFAULT_DEBUGGER_PORT;
+    this.verbose = options.verbose || false;
+  }
+
+  connect() {
+    if (this.socket) {
+      this.disconnect();
```

Comment:
Why disconnect if already connected?

Perhaps `return Promise.resolve()`?

Reply:
This does create a (very?) minor bug though, because if connect() is called before the socket has finished connecting the resolving promise indicates it has connected. Maybe better to throw as it should never happen anyway?

Reply:
Oh, shoot I only saw this after (squash-)merging.

Good catch. Better to this assign the already created promise to an instance variable and return that instead of creating a new promise.